stardog/admin.py

```diff
@@ -484,36 +557,121 @@ def new_virtual_graph(self, name, mappings, options=None, datasource=None, db=No
 
         Examples:
             >>> admin.new_virtual_graph(
+                  'users', MappingFile('mappings.sms'), None, 'my_datasource'
+                )
+            >>> admin.new_virtual_graph(
+                  'users', MappingFile('mappings.ttl','SMS2'), None, 'my_datasource'
+                )
+            >>> admin.new_virtual_graph(  #DEPRECATED
                   'users', File('mappings.ttl'),
                   {'jdbc.driver': 'com.mysql.jdbc.Driver'}
                 )
         """
 
         if mappings:
-            with mappings.data() as data:
-                mappings = data.read().decode() if hasattr(data, "read") else data
+            if hasattr(mappings, "syntax") and mappings.syntax:
+                if options:
+                    options["mappings.syntax"] = mappings.syntax
+                else:
+                    options = {"mappings.syntax": mappings.syntax}
+            else:
+                if options:
+                    options["mappings.syntax"] = "STARDOG"
+                else:
+                    options = {
+                        "mappings.syntax": "STARDOG"
+                    }  # this is the default of the original method
 
-        meta = {}
-        meta["name"] = name
-        meta["mappings"] = mappings
-        if options is not None:
+            with mappings.data() as data:
+                if hasattr(data, "read"):
+                    r = data.read()
+                    mappings = r.decode() if hasattr(r, "decode") else r
+                else:
+                    mappings = data
+
+        meta = {"name": name, "mappings": mappings}
+        if options:
             meta["options"] = options
-        if datasource is not None:
+
+        if datasource:
             meta["data_source"] = datasource
-        if db is not None:
+
+        if db:
             meta["db"] = db
 
         self.client.post("/admin/virtual_graphs", json=meta)
         return VirtualGraph(name, self.client)
 
-    # TODO
-    # def virtual_graph_import_file(self):
-    #     """
-    #     Import a CSV or JSON file as a virtual graph into the knowledge graph.
-    #     https://stardog-union.github.io/http-docs/#operation/importFileVG
-    #     :return:
-    #     """
-    #     return
+    def import_file(self, db, mappings, input_file, options=None, named_graph=None):
+        """Import a JSON or CSV file.
+
+        Args:
+          db (str): Name of the database to import the data
+          mappings (MappingRaw or MappingFile): New mapping contents.
+          input_file(ImportFile or ImportRaw):
+          options (dict, Optional): Options for the new csv import.
+          named_graph (str, Optional): The namegraph to associate it too
+
+        Returns:
+            r.ok
+
+        Examples:
+            >>> admin.import_file(
+                  'mydb', File('mappings.ttl'),
+                  'test.csv'
+                )
+        """
+
+        if mappings:
```

Comment:
same as before, it's better practice to set the defaults to None, and then check `if mappings is not None`. Since it doesn't really affect here, because none of the python empty containers are a valid mappings file, it doesn't add much value so im not blocking on this